### DISCUSSION TITLE
Timezone in ELSA 3.0

### DISCUSSION THREAD

**Discussion by Manali-23** [elsa-workflows/elsa-core#4265] [Q&A]
I am using ELSA 3.0. I am looking for setting a timezone in ELSA. But not able to find that option.
From one of the old posts of @sfmskywalker, you mentioned that the timezone feature has been added to ELSA 3.0.
Can you please tell me the steps? where I will be able to set that

**Comment by sfmskywalker** [comment]
Elsa 3 does't have timezone implemented yet, so it's good that you created this ~~issue~~ discussion 👍🏻 
I will create an issue for it.

**Comment by Manali-23** [comment]
Got it. Thank you for confirming @sfmskywalker

**Comment by dev-thinks** [comment]
any updates on this. would like to have timezone injected in (for SystemClock) especially for Cron expressions with specific times.

**Reply by Nekrosis1** [reply]
Timezones have been implemented
you can inject
`Elsa.Studio.Localization.Time.ITimeZoneProvider`
with a timezone provider like:
`Elsa.Studio.Localization.Time.Providers.LocalTimeZoneProvider`
(local time)
something like:
`builder.Services.AddScoped<ITimeZoneProvider, LocalTimeZoneProvider>();`

**Reply by dev-thinks** [reply]
Thank you, I will give a try

**Comment by sfmskywalker** [comment]
For runtime scheduling, Elsa currently evaluates cron expressions in UTC; there is no built-in per-workflow or host time-zone setting for `SystemClock`/the scheduling parser.

The current implementation makes this explicit:

- `Elsa.Common.Services.SystemClock` exposes `DateTimeOffset.UtcNow`.
- `Elsa.Scheduling.Services.CronosCronParser.GetNextOccurrence` calls Cronos with `TimeZoneInfo.Utc`.
- `Elsa.Common.RecurringTasks.CronSchedule` also calculates its next occurrence from the UTC system clock.

So the `ITimeZoneProvider` / `LocalTimeZoneProvider` registrations mentioned above are for Elsa Studio's display formatting; they do not change server-side cron evaluation.

For a schedule that must run at a local civil time (and follow daylight-saving rules), calculate the required UTC schedule in the host/application layer, or use a scheduler/provider that explicitly supports the desired time zone. There is not currently an Elsa scheduling configuration switch that changes the parser to an arbitrary `TimeZoneInfo`.
